conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java

```diff
@@ -139,24 +145,68 @@ public void sendCustomPlaybackEvent(String name, Map<String, Object> attributes)
         }
     }
 
-    private void customEvent(BitmovinPlayerEvent event) {
-        customEvent(event, new HashMap<String, Object>());
+    /**
+     * Initializes a new conviva tracking session.
+     * <p>
+     * Warning: The integration can only be validated without external session managing. So when using this method we can
+     * no longer ensure that the session is managed at the correct time. Additional: Since some metadata attributes
+     * relies on the players source we can't ensure that all metadata attributes are present at session creation.
+     * Therefore it could be that there will be a 'ContentMetadata created late' issue after conviva validation.
+     * <p>
+     * If no source was loaded this method will throw an error.
+     */
+    public void initializeSession() throws ConvivaAnalyticsException {
+        initializeSession(null);
     }
 
-    private void customEvent(BitmovinPlayerEvent event, Map<String, Object> attributes) {
-        if (!isValidSession()) {
+    /**
+     * Initializes a new conviva tracking session.
+     * <p>
+     * Warning: The integration can only be validated without external session managing. So when using this method we can
+     * no longer ensure that the session is managed at the correct time. Additional: Since some metadata attributes
+     * relies on the players source we can't ensure that all metadata attributes are present at session creation.
+     * Therefore it could be that there will be a 'ContentMetadata created late' issue after conviva validation.
+     *
+     * @param assetName Will be used as contentMetadata.assetName if no source was loaded before. This overrides the
+     *                  source assetName. If no source was loaded and no assetName is present this method will throw an
+     *                  error.
+     */
+    public void initializeSession(String assetName) throws ConvivaAnalyticsException {
+        if (isSessionActive()) {
             return;
         }
 
-        String eventName = event.getClass().getSimpleName();
-        sendCustomPlaybackEvent("on" + eventName, attributes);
+        if (bitmovinPlayer.getConfig().getSourceItem() == null && assetName == null) {
+            throw new ConvivaAnalyticsException("AssetName is missing. Provide assetName attribute or load player source first");
+        }
+
+        if (assetName != null) {
+            metadataOverrides.assetName = assetName;
```

Comment:
You set the `assetName`, but you never reset it to `null`.
What if I don't want to override the `assetName` in a second `initializeSession` -> Currently, the old name would be kept.

Reply:
Good point. Need to do something here. (Same goes for iOS and Web)